cmd/koord-runtime-proxy/main.go

```diff
@@ -43,12 +43,13 @@ func main() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 	pflag.Parse()
 
-	dir, _ := filepath.Split(options.RuntimeProxyEndpoint)
-	err := os.MkdirAll(dir, 0777)
-	if err != nil {
-		klog.Fatalf("Failed to create socket dir, err: %v", err)
+	if err := os.Remove(options.RuntimeProxyEndpoint); err != nil && !os.IsNotExist(err) {
```

Comment:
Could we use syscall.Unlink() to achieve it? like https://github.com/koordinator-sh/koordinator/blob/0072d6bf8c2edee509340e644e5808590bff3c4f/pkg/koordlet/runtimehooks/server/server.go#L74 ?

Reply:
> Could we use syscall.Unlink() to achieve it?

`os.Remove` will call either `syscall.Unlink` or `syscall.Rmdir` under different circumstances, so I would like to leave the cleanup job to it.

> what if user wrongly used an already used socket file by mistake, doing so could delete the socket file by force right? Is this allowed?

It's fine.